php/OAuthSimple.php

```diff
@@ -490,9 +490,16 @@ private function _generateSignature ($parameters="")
 		{
             $secretKey .= self::_oauthEscape($this->_secrets['oauth_secret']);
         }
-        if(empty($parameters)){
-            $parameters = $this->_normalizedParameters();
+        
+        if(empty($parameters))
+        {
+            $parameters = urlencode($this->_normalizedParameters());
+        } 
+        else
+        {
+            $parameters = urlencode($parameters);
```

Comment:
I looked in to the code - there is just one place where you are invoking _generateSignature and you do this with the valid parameter. The code affected by my change should actually look like this: 
if(!empty($parameters))
 {
$parameters = urlencode($parameters);
} 
the original version did not make sense:
if(empty($parameters)){
$parameters = $this->_normalizedParameters();
}
because it is executing _normalizedParameters() again even though it had been executed just a moment before(on line 286). To make it right we have 2 alternative solutions:
1) remove $parameters from _generateSignature ($parameters="") method signature and replace edited part with just one line:
$parameters = urlencode($this->_normalizedParameters());
or
2) replace just edited block with:
if(!empty($parameters))
 {
$parameters = urlencode($parameters);
}